transport/grpc/integration_test.go

```diff
@@ -170,6 +170,17 @@ func TestYARPCMaxMsgSize(t *testing.T) {
 	})
 }
 
+func TestDataRace(t *testing.T) {
```

Comment:
does this test fail without the fix?

Reply:
It does not...  I don't think we'll be able to get significant pooling tests without #1408, I'll leave the test, but it's mostly just "large" test.